csm/common/filter.py

```diff
@@ -0,0 +1,90 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from cortx.utils.data.access.filters import Compare, And, Or
+from schematics.exceptions import ConversionError
+from csm.core.blogic.models import CsmModel
+from csm.common.errors import InvalidRequest
+import urllib.parse
+import re
+
+class Filter:
+    """Query to filter_object builder"""
+
+    def _parse_query(query):
+        """
+        Segregate list of key_val pairs and operands.
+        :param query: string of key_val pairs and operands like: "key1=value1 AND key2=value2 OR key3=value3"
+        :returns: dict(key_value_fields), dictionary of key and value pairs
+        :returns: operations, Ordered list of operations to be performed on give field and value pairs
+        """
+        operations = []
+        key_value_fields = []
+        if query:
+            fields = list(re.split(' OR | AND ', query))
+            key_value_fields = [(map(str.strip, key.split('=', 1))) for key in fields]
+            fields = list(query.split(" "))
+            operations = [key for key in fields if key in ("AND", "OR")]
+        return dict(key_value_fields), operations
+
+    def _validate_query_fields(fields, model):
+        """
+        Validate and provide default values for those keys whose values are invalid for given field.
+        :param fields: collection of key value pairs received as input
+        :param Type[BaseModel] model: model for constructing data mapping for index
+        :returns: Updated key_value pairs
+        """
+        default_val_dict = model().to_native()
+        updated_key_val = {}
+        for key, value in fields.items():
+            try:
+                field = eval(f"model.{key}")
+                valid_operand = field.to_native(value)
+            except AttributeError as err:
+                raise InvalidRequest(f"{key} not found: {err}")
```

Comment:
error should not have internal details like CSMAuditLogModel

Reply:
I have addressed it in following revision of pull request: [583](https://github.com/Seagate/cortx-manager/pull/583)